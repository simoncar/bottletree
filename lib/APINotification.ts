import {
  addDoc,
  doc,
  updateDoc,
  collection,
  onSnapshot,
  setDoc,
  Timestamp,
  getDocs,
  documentId,
  orderBy,
} from "firebase/firestore";
import { auth, db } from "./firebase";
import { IPushToken, IUser } from "./types";
import * as Notifications from "expo-notifications";

// ExponentPushToken[z-50OyGeRPth6nxZSWk_A4]

export function setToken(
  token: IPushToken,
  callback: { (id: string): void; (arg0: string): void },
) {
  try {
    const docRef = doc(db, "tokens", token.key);

    console.log(
      "setToken",
      token.key,
      token.pushToken,
      auth.currentUser?.uid,
      auth.currentUser?.displayName,
    );

    setDoc(
      docRef,
      {
        pushToken: token.pushToken,
        uid: auth.currentUser?.uid,
        displayName: auth.currentUser?.displayName,
        timestamp: Timestamp.now(),
      },
      { merge: true },
    ).then((docRef) => {
      callback(token.key);
    });
  } catch (e) {
    console.error("Error adding token: ", e);
  }

  return;
}

export async function getToken() {
  const token = (await Notifications.getExpoPushTokenAsync()).data;
  console.log("UUUUUU:", token);
  return token;
}
