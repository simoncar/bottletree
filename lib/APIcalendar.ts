import { addDoc, collection, onSnapshot, query } from "firebase/firestore";
import { db } from "./firebase";
import { IProject, ICalendarEvent } from "./types";

type projectsRead = (projects: IProject[]) => void;

export async function getItems(projectId: string, callback: itemsRead) {
    const q = query(collection(db, "calendar"));

    const unsubscribe = onSnapshot(q, (querySnapshot) => {
        const calendarEvents: ICalendarEvent[] = [];
        querySnapshot.forEach((doc) => {
            calendarEvents.push({
                key: doc.id,
                allDay: doc.data().allDay,
                dateBegin: doc.data().dateBegin,
                dateEnd: doc.data().dateEnd,
                description: doc.data().description,
                projectId: doc.data().projectId,
                title: doc.data().title,
                uid: doc.data().uid,
                dateBeginSplit: doc
                    .data()
                    .dateBegin.toDate()
                    .toISOString()
                    .split("T")[0],
                dateEndSplit: doc
                    .data()
                    .dateEnd.toDate()
                    .toISOString()
                    .split("T")[0],
                timeBegin: doc
                    .data()
                    .dateBegin.toDate()
                    .toLocaleTimeString("en-US"),
                timeEnd: doc
                    .data()
                    .dateEnd.toDate()
                    .toLocaleTimeString("en-US"),
            });
        });

        const expandedDates = expandDateRanges(calendarEvents);
        console.log("expandedDates", expandedDates);

        callback(expandedDates);
    });

    return () => unsubscribe();
}

interface DateRange {
    dateBegin: string;
    dateEnd: string;
}

interface ExpandedDates {
    [date: string]: string;
}

function expandDateRanges(dateRanges: DateRange[]): ExpandedDates {
    const expandedDates: ExpandedDates = {};

    for (const range of dateRanges) {
        const startDate = new Date(range.dateBeginSplit);
        const endDate = new Date(range.dateEndSplit);

        const currentDate = new Date(startDate);
        while (currentDate <= endDate) {
            const dateString = currentDate.toISOString().split("T")[0];
            expandedDates[dateString] = [
                ...(expandedDates[dateString] || []),
                {
                    title: range.title,
                    description: range.description,
                    projectId: range.projectId,
                    key: range.key,
                    dateBegin: range.dateBegin,
                    dateEnd: range.dateEnd,
                    allDay: range.allDay,
                    uid: range.uid,
                    dateBeginSplit: range.dateBeginSplit,
                    timeBegin: range.timeBegin,
                    timeEnd: range.timeEnd,
                },
            ];

            currentDate.setDate(currentDate.getDate() + 1);
        }
    }

    return expandedDates;
}

export function addCalendarEvent(
    calendarEvent: ICalendarEvent,
    callback: { (id: string): void; (arg0: string): void },
) {
    try {
        addDoc(collection(db, "calendar"), {
            allDay: calendarEvent.allDay,
            dateBegin: calendarEvent.dateBegin,
            dateEnd: calendarEvent.dateEnd,
            description: calendarEvent.description,
            projectId: calendarEvent.projectId,
            title: calendarEvent.title,
            uid: calendarEvent.uid,
        }).then((docRef) => {
            callback(docRef.id);
        });
    } catch (e) {
        console.error("Error adding calendar event: ", e);
    }

    return;
}
