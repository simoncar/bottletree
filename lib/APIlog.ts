import * as Application from "expo-application";
import * as Device from "expo-device";
import { dbm } from "./firebase";
import { ILog } from "./types";

import {
  addDoc,
  collection,
  getDocs,
  orderBy,
  query,
  serverTimestamp,
} from "@react-native-firebase/firestore";

// Update getLogs to use modular API
export async function getLogs(callback: logsRead) {
  const logs: ILog[] = [];
  const logsQuery = query(
    collection(dbm, "logs"),
    orderBy("timestamp", "desc"),
  );
  const logSnapshot = await getDocs(logsQuery);

  logSnapshot.forEach((doc) => {
    const data = doc.data();
    logs.push({
      key: doc.id,
      timestamp: data.timestamp,
      loglevel: data.loglevel,
      message: data.message,
      user: data.user,
      device: data.device,
      version: data.version,
      email: data.email,
    });
  });

  callback(logs);
}

// Update addLog to use modular API
export async function addLog(log: ILog) {
  log.timestamp = serverTimestamp();
  log.device =
    Device.manufacturer +
    " > " +
    Device.modelName +
    " > " +
    Device.osName +
    " > " +
    Device.osVersion;
  log.version = Application.nativeApplicationVersion;

  try {
    const docRef = await addDoc(collection(dbm, "logs"), log);
    console.log("Log written with ID: ", docRef.id);
  } catch (e) {
    console.error("Error adding log: ", e);
  }
}
