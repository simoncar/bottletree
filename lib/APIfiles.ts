import { dbm } from "@/lib/firebase"; // or "@/lib/firebase.web"
import {
  addDoc,
  collection,
  doc,
  onSnapshot,
  orderBy,
  query,
  serverTimestamp,
} from "@react-native-firebase/firestore";

// import {
//   getDownloadURL,
//   getStorage,
//   ref as storageRef,
// } from "@react-native-firebase/storage";
import storage, { getDownloadURL } from "@react-native-firebase/storage";

import * as Crypto from "expo-crypto";
import * as DocumentPicker from "expo-document-picker";
import { setPostFile } from "./APIpost";
import { IFile, IPost, IUser } from "./types";

type FileData = {
  mimeType: string;
  name: string;
  size: number;
  uri: string;
};

type FilesJson = {
  assets: FileData[];
  canceled: boolean;
};

export async function deleteFile(
  projectId: string,
  key: string,
  callback: (id: string) => void,
) {
  try {
    const docRef = doc(
      collection(doc(collection(dbm, "projects"), projectId), "files"),
      key,
    );

    await dbm.runTransaction(async (transaction) => {
      transaction.delete(docRef);
    });

    callback(key);
  } catch (e) {
    console.error("Error deleting file from project: ", e);
  }
}
export async function getFiles(project: string, callback: filesRead) {
  const filesRef = collection(
    doc(collection(dbm, "projects"), project),
    "files",
  );
  const q = query(filesRef, orderBy("modified", "desc"));

  const unsubscribe = onSnapshot(q, (querySnapshot) => {
    const files: IFile[] = [];

    querySnapshot.forEach((docSnap) => {
      const data = docSnap.data();
      files.push({
        key: docSnap.id,
        filename: data.filename,
        url: data.url,
        mimeType: data.mimeType,
        bytes: data.bytes,
        created: data.created,
        modified: data.modified,
        projectId: data.projectId,
      });
    });

    callback(files);
  });

  return unsubscribe;
}

export async function uploadFilesAndCreateEntries(
  filesJson: DocumentPicker.DocumentPickerResult,
  projectId: string,
  user: IUser,
) {
  if (filesJson.canceled) {
    console.log("File selection was canceled.");
    return;
  }

  const projectFilesRef = collection(doc(dbm, "projects", projectId), "files");
  console.log("uploadFilesAndCreateEntries - modular:", filesJson);

  for (const file of filesJson.assets) {
    try {
      const UUID = Crypto.randomUUID();
      //const blob = await uriToBlob(file.uri);

      const filePath = `project/${projectId}/files/${UUID}`;
      const fileStorageRef = storage().ref(filePath);

      try {
        console.log("AAA111- fileStorageRef.putFile");
        await fileStorageRef.putFile(file.uri);
        console.log("AAA222- fileStorageRef.putFile");
        //await uploadBytes(fileStorageRef, blob);
      } catch (error: any) {
        if (error.code === "storage/unauthorized") {
          console.error("User does not have permission to access the object");
        } else {
          throw error;
        }
      }

      const downloadURL = await getDownloadURL(fileStorageRef);

      const fileEntry: IFile = {
        key: UUID,
        filename: file.name,
        url: downloadURL,
        mimeType: file.mimeType,
        bytes: file.size,
        created: serverTimestamp(),
        modified: serverTimestamp(),
        projectId,
      };

      await addDoc(projectFilesRef, fileEntry);

      const post: IPost = {
        key: UUID,
        caption: `${file.name}`,
        projectId,
        author: user.displayName,
        uid: user.uid,
        timestamp: serverTimestamp(),
        ratio: 1,
        file: UUID,
      };

      setPostFile(post, (postId) => {
        console.log(`Post created with ID: ${postId}`);
      });

      console.log(`File ${file.name} uploaded and entry created in Firestore.`);
    } catch (error) {
      console.error(`Error uploading file ${file.name}:`, error);
    }
  }
}
