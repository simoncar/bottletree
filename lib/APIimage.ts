import * as ImagePicker from "expo-image-picker";
import { firebase, uploadBytes } from "@/lib/firebase";
import * as Crypto from "expo-crypto";
import { Platform } from "react-native";
import * as ImageManipulator from "expo-image-manipulator";

// Types and interfaces
interface ImageAsset {
  uri: string;
  height: number;
  width: number;
}

interface ImageProcessOptions {
  compress: number;
  format: ImageManipulator.SaveFormat;
  base64: boolean;
}

interface ProgressCallback {
  (progress: number): void;
}

// Constants
const DEFAULT_COMPRESSION_RATIO = 0.2;
const IMAGE_FORMAT = ImageManipulator.SaveFormat.JPEG;
const IS_WEB_PLATFORM = Platform.OS === "web";

export const addImageFromCameraRoll = async (
  multiple: boolean,
  folder: string,
  project: string,
  progressCallback: {
    (progress: any): void;
  },
  completedCallback: {
    (sourceDownloadURLarray: any): void;
    (arg0: unknown[]): void;
  },
) => {
  const result = await ImagePicker.launchImageLibraryAsync({
    mediaTypes: ImagePicker.MediaTypeOptions.Images,
    quality: 1,
    allowsMultipleSelection: multiple,
  });

  if (!result.canceled) {
    try {
      const promises = result.assets.map((item) =>
        processItemAsync(folder, project, item, progressCallback),
      );
      const processedResults = await Promise.all(promises);
      completedCallback(processedResults);
    } catch (error) {
      console.error("Error occurred during processing:", error);
    }
  }
};

export const addImageFromPhoto = async (
  photo: any,
  folder: string,
  project: string,
  progressCallback: (progress: any) => void,
  completedCallback: { (sourceDownloadURL: any): void; (arg0: unknown): void },
) => {
  try {
    const processedResults = await processItemAsync(
      folder,
      project,
      photo,
      progressCallback,
    );
    completedCallback(processedResults);
  } catch (error) {
    console.error("Error occurred during processing addImageFromPhoto:", error);
  }
};

async function processItemAsync(
  folder: string,
  project: string,
  asset: ImageAsset,
  progressCallback: ProgressCallback,
): Promise<string> {
  let imageUri = asset.uri;
  let compressedImage;

  try {
    console.log("processItemAsync: ", asset);
    console.log("processItemAsync: ", asset.uri);

    if (IS_WEB_PLATFORM) {
      // TODO: for some reason this is not working on web, results in a larger file size and PNG?
      compressedImage = asset;
    } else {
      const compressionOptions: ImageProcessOptions = {
        compress: DEFAULT_COMPRESSION_RATIO,
        format: IMAGE_FORMAT,
        base64: true,
      };

      compressedImage = await ImageManipulator.manipulateAsync(
        asset.uri,
        [], // No operations, just compression
        compressionOptions,
      );
      imageUri = compressedImage.uri;
    }
  } catch (error) {
    console.error("Error compressing image:", error);
    throw error;
  }

  const getBlobFromUri = async (uri: string): Promise<Blob> => {
    return new Promise((resolve, reject) => {
      const xhr = new XMLHttpRequest();
      xhr.onload = () => resolve(xhr.response);
      xhr.onerror = (e) =>
        reject(new TypeError(`Network request failed: ${e}`));
      xhr.responseType = "blob";
      xhr.open("GET", uri, true);
      xhr.send(null);
    });
  };

  return new Promise((resolve, reject) => {
    try {
      const storageRef = getStorageRef(folder, project);

      if (IS_WEB_PLATFORM) {
        getBlobFromUri(imageUri).then((blob) => {
          firebase
            .storage()
            .ref(storageRef.fullPath)
            .put(blob)
            .then(() => {
              storageRef.getDownloadURL().then((downloadURL) => {
                console.log("File available at", downloadURL);
                console.log("dimensions", asset.height, asset.width);
                const ratio = asset.height / asset.width;
                resolve(ratio + "*" + downloadURL);
              });
            });
        });
      } else {
        const uploadTask = storageRef.putFile(imageUri);

        uploadTask.on(
          "state_changed",
          (snapshot) => {
            const uploadProgress = Math.floor(
              (snapshot.bytesTransferred / snapshot.totalBytes) * 100,
            );
            progressCallback(uploadProgress);

            switch (snapshot.state) {
              case "paused":
                console.log("Upload is paused");
                break;
              case "running":
                console.log(
                  "Upload is running:",
                  snapshot.bytesTransferred,
                  snapshot.totalBytes,
                );
                break;
            }
          },
          (error) => {
            console.log("error file upload:", error);
          },
          () => {
            storageRef.getDownloadURL().then((downloadURL) => {
              console.log("File available at", downloadURL);
              console.log("dimensions", asset.height, asset.width);
              const ratio = asset.height / asset.width;
              resolve(ratio + "*" + downloadURL);
            });
          },
        );
      }
    } catch (error) {
      console.log("processItemAsync error:", error);
      reject(error);
    }
  });
}

function getStorageRef(folder: string, project: string) {
  const UUID = Crypto.randomUUID();
  const d = new Date();

  const fileName = folder + "/" + project + "/images/" + UUID;

  //const storageRef = ref(storage, fileName);
  const storageRef = firebase.storage().ref(fileName);
  console.log("storageRef", storageRef);

  return storageRef;
}
