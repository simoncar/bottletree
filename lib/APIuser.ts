import { CurrentRenderContext } from "@react-navigation/native";
import { auth, firestore } from "@/lib/firebase";
import { IUser } from "./types";

export async function getUser(
  uid: string,
  callback: { (user: IUser): void; (arg0: IUser): void },
) {
  const q = firestore().collection("users").doc(uid);

  q.get()
    .then((doc) => {
      if (doc.exists) {
        const user: IUser = {
          key: doc.id,
          uid: doc.id,
          displayName: doc.data()?.displayName,
          email: doc.data()?.email,
          photoURL: doc.data()?.photoURL,
          language: doc.data()?.language,
          project: doc.data()?.project,
          postCount: doc.data()?.postCount,
        };

        callback(user);
      } else {
        callback(null);
      }
    })
    .catch((error) => {
      console.log("getUser Error getting document:", error);
      callback(null);
    });

  return () => q;
}

export async function deleteUser(
  uid: string,
  callback: { (user: IUser): void; (arg0: IUser): void },
) {
  console.log("deleteUser: ", uid);

  const q = firestore().collection("users").doc(uid);

  q.delete()
    .then(() => {
      console.log("User successfully deleted!");
      callback(null);
    })
    .catch((error) => {
      console.error("Error removing user: ", error);
      callback(null);
    });

  return () => q;
}

export const updateAccountName = (displayName: string) => {
  const docRef1 = firestore().collection("users").doc(auth().currentUser.uid);

  const user = auth().currentUser;
  const docRef2 = user
    .updateProfile({
      displayName: displayName,
    })
    .then(() => {
      docRef1.set(
        {
          displayName: displayName,
          email: auth().currentUser.email,
          photoURL: auth().currentUser.photoURL,
        },
        { merge: true },
      );
    })
    .catch((error) => {
      console.log("upupdateAccountName update ERROR ", error);
    });
};

export const updateUserProjectCount = (project: string) => {
  const u = auth().currentUser?.uid;
  if (project === "welcome") {
    return;
  }

  const docRef = firestore().collection("projects").doc(project);
  let count = 0;
  docRef
    .get()
    .then((doc) => {
      if (doc.exists) {
        if (typeof doc.data().postCount === "number") {
          count = doc.data().postCount;
        }
      }
    })
    .then(() => {
      firestore()
        .collection("users")
        .doc(u)
        .set(
          {
            project: project,
            postCount: {
              [project]: count,
            },
          },
          { merge: true },
        )
        .then(() => {
          console.log("User's project count updated successfully!");
        })
        .catch((error) => {
          console.log("Error updating user's project count:", error);
        });
    });
};

export const updateAccountPhotoURL = (photoURL: string) => {
  const a = auth().currentUser;
  if (a) {
    const docRef1 = firestore().collection("users").doc(a?.uid);

    const user = auth().currentUser;

    user
      .updateProfile({
        photoURL: photoURL,
      })
      .then(() => {
        docRef1.set(
          {
            photoURL: photoURL,
            email: auth().currentUser.email,
            displayName: auth().currentUser.displayName,
          },
          { merge: true },
        );
      })
      .then(() => {})
      .catch((error) => {
        console.log("updateAccountPhotoURL update ERROR ", error);
      });
  }
};

export async function getUsers(callback: usersRead) {
  const users: IUser[] = [];

  const q = firestore().collection("users").orderBy("displayName", "asc");

  const usersSnapshot = await q.get();

  usersSnapshot.forEach((doc) => {
    users.push({
      key: doc.id,
      uid: doc.id,
      displayName: doc.data().displayName,
      email: doc.data().email,
      photoURL: doc.data().photoURL,
      language: doc.data().language,
      project: doc.data().project,
    });
  });

  callback(users);
}

const generateFirebaseDocIDFromEmail = (email) => {
  if (!email || typeof email !== "string") {
    throw new Error("Invalid email provided");
  }

  // Replace invalid characters for Firebase keys
  const sanitizedEmail = email.replace(/[.$#\[\]\/]/g, "_");

  // Add a random suffix to ensure uniqueness
  const suffix = Math.random().toString(36).substring(2, 8);

  return `${sanitizedEmail}_${suffix}`;
};

export async function getUserProjectCount(callback: userProjectCountRead) {
  let user: IUser;

  firestore()
    .collection("users")
    .doc(auth().currentUser.uid)
    .get()
    .then((doc) => {
      user = {
        key: doc.id,
        uid: doc.id,
        displayName: doc.data()?.displayName,
        email: doc.data()?.email,
        photoURL: doc.data()?.photoURL,
        postCount: doc.data()?.postCount,
        language: doc.data()?.language,
        project: doc.data()?.project,
      };

      callback(user);
    });
}
