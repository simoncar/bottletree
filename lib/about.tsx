import React, { useContext, useState, useEffect } from "react";
import { Button, Pressable, StyleSheet } from "react-native";
import { Text, View } from "@/components/Themed";
import { useSession } from "@/lib/ctx";
import { auth } from "@/lib/firebase";
import { UserContext } from "@/lib/UserContext";
import { Link } from "expo-router";
import * as Application from "expo-application";
import * as Sentry from "@sentry/react-native";
import { useRouter } from "expo-router";

export const About = () => {
  const { user } = useContext(UserContext);
  const { session } = useSession();
  const [authUser, setAuthUser] = useState<string>(null);

  useEffect(() => {
    setAuthUser(auth().currentUser?.uid);
  }, []);

  const router = useRouter();
  let pressCount = 0;

  const handlePress = () => {
    pressCount += 1;
    console.log(pressCount);

    if (pressCount === 3) {
      router.push("/p");
    }
  };

  return (
    <View style={styles.aboutContainer}>
      <Pressable onPress={handlePress}>
        <Text style={styles.version}>One Build</Text>
      </Pressable>

      <Text style={styles.version}>
        {Application.nativeApplicationVersion} ({Application.nativeBuildVersion}
        ) | 12.app.sentry
      </Text>
      <Text style={styles.version}>Email: {user?.email}</Text>
      <Text style={styles.version}>Name: {user?.displayName}</Text>
      <Text style={styles.version}>User Context - {user?.uid}</Text>
      <Text style={styles.version}>Session - {session}</Text>
      <Text style={styles.version}>Auth - {auth().currentUser?.uid}</Text>
    </View>
  );
};

const styles = StyleSheet.create({
  aboutContainer: {
    alignItems: "center",
    justifyContent: "center",
    paddingBottom: 50,
    paddingTop: 50,
  },
  version: {
    color: "grey",
    fontSize: 10,
  },
});
