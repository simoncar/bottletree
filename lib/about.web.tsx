import React, { useState, useEffect, useContext } from "react";
import { StyleSheet } from "react-native";
import { getLocales } from "expo-localization";
import { Text, View } from "@/components/Themed";
import * as Application from "expo-application";
import { useSession } from "@/lib/ctx";
import { auth } from "@/lib/firebase";
import { UserContext } from "@/lib/UserContext";

export const About = () => {
  const { user } = useContext(UserContext);
  const { session } = useSession();
  const [authUser, setAuthUser] = useState<string>(true);
  const deviceLanguage = getLocales()[0].languageCode;

  useEffect(() => {
    setAuthUser(auth().currentUser?.uid);
  }, []);

  if (null == user) {
    return <View></View>;
  } else {
    return (
      <View style={styles.aboutContainer}>
        <Text style={styles.version}>{user.displayName}</Text>
        <Text style={styles.version}>{user.email}</Text>
        <Text style={styles.version}>
          Anonymous: {auth().currentUser?.isAnonymous}
        </Text>
        <Text style={styles.version}>
          {Application.nativeApplicationVersion} (
          {Application.nativeBuildVersion}) | 2
        </Text>

        <Text style={styles.version}>User Context - {user.uid}</Text>
        <Text style={styles.version}>CTX - {session}</Text>
        <Text style={styles.version}>Auth2 - {authUser}</Text>
        <Text style={styles.version}>Language - {deviceLanguage}</Text>
        <Text style={styles.version}>Project - {user.project}</Text>
      </View>
    );
  }
};

const styles = StyleSheet.create({
  aboutContainer: {
    alignItems: "center",
    justifyContent: "center",
    paddingTop: 50,
  },
  version: {
    color: "grey",
    fontSize: 14,
  },
});
