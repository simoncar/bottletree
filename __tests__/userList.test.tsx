import React from "react";
import UserList from "../app/(app)/userList";
import { getUsers } from "@/lib/APIuser";
import { renderRouter, screen } from "expo-router/testing-library";

jest.mock("@/lib/APIuser", () => ({
  getUsers: jest.fn(),
}));

it("user list test", async () => {
  const MockComponent = jest.fn(() => <UserList />);

  renderRouter(
    {
      index: MockComponent,
      "note?projectId=73JwAXeOEhLXUggpVKK9": MockComponent,
      "(group)/b": MockComponent,
    },
    {
      initialUrl: "/",
    },
  );

  expect(screen).toHavePathname("/");
  expect(screen).toMatchSnapshot();
});
