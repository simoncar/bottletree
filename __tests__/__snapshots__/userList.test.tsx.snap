// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`user list test 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    [
      {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <RNCSafeAreaView
    edges={
      {
        "bottom": "additive",
        "left": "additive",
        "right": "additive",
        "top": "additive",
      }
    }
    style={
      {
        "flex": 1,
      }
    }
  >
    <View
      style={
        {
          "flex": 1,
          "height": 200,
        }
      }
    >
      <TextInput
        onChangeText={[Function]}
        placeholder="Search ..."
        placeholderTextColor="#ccc"
        style={
          [
            {
              "color": "#000",
            },
            {
              "borderColor": "gray",
              "borderRadius": 5,
              "borderWidth": 1,
              "height": 40,
              "margin": 10,
              "paddingHorizontal": 8,
            },
          ]
        }
        value=""
      />
      <RCTScrollView
        style={
          {
            "paddingBottom": 50,
          }
        }
      >
        <View />
      </RCTScrollView>
    </View>
  </RNCSafeAreaView>
</RNCSafeAreaProvider>
`;
